### PR TITLE
Adjust docu to PR #943 and more on graded rings

### PR DESCRIPTION
**Pull request by wdecker**

### CODE REVIEW

File: docs/src/CommutativeAlgebra/rings.md [hunk 1]
```diff
@@ -201,11 +243,27 @@ g = R(QQ.([3, 1]), [[2, 0, 0], [0, 1, 1]])
...
 
-!!! note
-    An often more effective way to create polynomials is to use the `MPoly` build context, see the chapter on rings.
+An often more effective way to create polynomials is to use the `MPoly` build context as indicated below:
+
+```@repl oscar
+R, (x, y) = PolynomialRing(QQ, ["x", "y"])
+B = MPolyBuildCtx(R)
+for i = 1:5 push_term!(B, QQ(i), [i, i-1]) end
+finish(B::MPolyBuildCtx)
```
Comment (thofma) [inline docs/src/CommutativeAlgebra/rings.md]:
```suggestion
finish(B)
```

File: src/Rings/mpoly-graded.jl [hunk 2]
```diff
@@ -367,6 +583,7 @@ function addeq!(a::MPolyElem_dec, b::MPolyElem_dec)
 end
 
 length(a::MPolyElem_dec) = length(a.f)
+total_degree(a::MPolyElem_dec) = total_degree(a.f)
```
Comment (thofma) [inline src/Rings/mpoly-graded.jl:627]: I am not an expert, but here the degree is not related to the grading? Just curiosity.

Comment (wdecker) [reply src/Rings/mpoly-graded.jl:627]: I understand that the total_degree function returns the total degree of a polynomial with respect to the standard grading. The degree function returns the degree of a homogeneous polynomial with respect to the given grading. If this is confusing, let us reconsider ...

Comment (thofma) [reply src/Rings/mpoly-graded.jl:627]: No, it is fine with me. I am happy to keep it.

File: docs/src/CommutativeAlgebra/rings.md [hunk 3]
```diff
@@ -128,26 +129,57 @@ QT = FractionField(T)
...
-GradedPolynomialRing(C::Ring, V::Vector{String}, W::Vector{Int}; ordering=:lex)
+grade(R::MPolyRing, W::Union{fmpz_mat, Matrix{<:IntegerUnion}})
 ```
 
-!!! note
-    OSCAR functionality for working with gradings by arbitrary abelian groups and for working with filtrations is currently under development.
+```@docs
+grade(R::MPolyRing, W::Vector{<:IntegerUnion})
+```
+:
```
Comment (fingolfin) [inline docs/src/CommutativeAlgebra/rings.md]:
```suggestion

```

**Review by fingolfin — COMMENTED** [review]
I'll look through this later, but quick question: shall I push a doctest fix for you?

File: docs/src/CommutativeAlgebra/rings.md (hunk 3 above)
Comment (wdecker) [reply docs/src/CommutativeAlgebra/rings.md]: No, thanks, I saw this too late ...

**Review by fingolfin — APPROVED** [review]
Let's squash merge once tests completed

**Pull request merged**